### PR TITLE
Make adaptive form

### PR DESCRIPTION
**Pull request by Benjtalkshow**
#156 
This is a draft PR. I encountered significant draw-back when i synced with the main. i am fixing it right away. Then, work on the footer.

### CODE REVIEW

**Comment by Benjtalkshow** [comment]
Hello @djeck1432 
I just made another changes. I want to adjust the links and buttons on the sidebar and make the final changes. Thanks

**Comment by djeck1432** [comment]
@Benjtalkshow What is the status of this task?

**Comment by Benjtalkshow** [comment]
> @Benjtalkshow What is the status of this task?

I just made my final PR. Review it and give me feedback ser.

**Comment by Benjtalkshow** [comment]
> @Benjtalkshow What is the status of this task?

I just made my final PR. Review it and give me feedback ser.

File: frontend/src/components/header/Header.jsx [hunk 1]
```diff
@@ -1,44 +1,90 @@
...
+      ) {
+        setIsMenuOpen(false);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:30]: Please create separate custom hook `useClickOutside` and use it instead

File: frontend/src/components/BalanceCards.jsx [hunk 2]
```diff
@@ -6,32 +6,56 @@ import { ReactComponent as DAI } from '../assets/icons/dai.svg';
...
+    const handleMediaQueryChange = (e) => {
+      setIsMobile(e.matches);
+    };
+
+    mediaQuery.addEventListener('change', handleMediaQueryChange);
+
+    return () => {
+      mediaQuery.removeEventListener('change', handleMediaQueryChange);
+    };
+  }, []);
```
Comment (whateverfw) [inline frontend/src/components/BalanceCards.jsx:33]: Please create separate custom hook `useMatchMedia` and use it instead

File: frontend/src/pages/forms/form.css [hunk 3]
```diff
@@ -325,3 +326,41 @@ input[type='number'].error {
...
+  .balance-item {
+    height: fit-content;
+    padding: 1rem 0rem;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    gap: 1rem;
+  }
+.balance-item .balance-title {
+  font-size: 1.2rem!important;
```
Comment (whateverfw) [inline frontend/src/pages/forms/form.css:347]: avoid important

File: frontend/src/pages/forms/form.css [hunk 4]
```diff
@@ -325,3 +326,41 @@ input[type='number'].error {
...
+  display: flex;
+  align-items: center;
+}
+.balance-item label {
+  font-size: 1rem!important;
```
Comment (whateverfw) [inline frontend/src/pages/forms/form.css:352]: avoid important

File: frontend/src/components/header/Header.jsx [hunk 5]
```diff
@@ -1,44 +1,90 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { ReactComponent as Logo } from 'assets/images/logo.svg';
+import { ReactComponent as Menu } from 'assets/images/menu.svg';
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:3]: this line break build, seems like `menu.svg` doesn't exist at all, could you please double check and fix?

File: frontend/src/components/header/Header.jsx [hunk 6]
```diff
@@ -1,44 +1,90 @@
...
+import { ReactComponent as Close } from 'assets/images/close.svg';
```
Comment (whateverfw) [inline frontend/src/components/header/Header.jsx:4]: also please verify that this one exists

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/components/header/Header.jsx (hunk 1 above)
Comment (Benjtalkshow) [reply frontend/src/components/header/Header.jsx:30]: Alright

File: frontend/src/components/BalanceCards.jsx (hunk 2 above)
Comment (Benjtalkshow) [reply frontend/src/components/BalanceCards.jsx:33]: Alright

File: frontend/src/pages/forms/form.css (hunk 3 above)
Comment (Benjtalkshow) [reply frontend/src/pages/forms/form.css:347]: ok

File: frontend/src/pages/forms/form.css (hunk 4 above)
Comment (Benjtalkshow) [reply frontend/src/pages/forms/form.css:352]: ok

File: frontend/src/components/header/Header.jsx (hunk 5 above)
Comment (Benjtalkshow) [reply frontend/src/components/header/Header.jsx:3]: alright

File: frontend/src/components/header/Header.jsx (hunk 6 above)
Comment (Benjtalkshow) [reply frontend/src/components/header/Header.jsx:4]: alright

File: frontend/src/components/header/Header.jsx (hunk 5 above)
Comment (Benjtalkshow) [reply frontend/src/components/header/Header.jsx:3]: its breaking insides assets/images so i have moved it to assets/icons

File: frontend/src/components/header/Header.jsx (hunk 6 above)
Comment (Benjtalkshow) [reply frontend/src/components/header/Header.jsx:4]: its breaking insides assets/images so i have moved it to assets/icons

**Comment by Benjtalkshow** [comment]
I have made the requested changes. @djeck1432

**Review by whateverfw — APPROVED** [review]
Seems good to me @Benjtalkshow @djeck1432

**Comment by djeck1432** [comment]
@Benjtalkshow Please, resolve conflicts and I will merge it

**Review by whateverfw — CHANGES_REQUESTED** [review]
@Benjtalkshow Could you please also resolve merge conflicts with main (and use footer and header versions from there and make sure everything works correctly along with your changes after the merge). Thanks!

**Comment by djeck1432** [comment]
@Benjtalkshow Please, resolve merge conflict

**Comment by Benjtalkshow** [comment]
@djeck1432  @whateverfw 
merge conflict solved

**Review by whateverfw — APPROVED** [review]
Looks good, thanks!

**Review by djeck1432 — APPROVED** [review]
Good job, looks good to me

**Pull request merged**